src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/MigrateBookingIntegrationTest.kt

```diff
@@ -0,0 +1,328 @@
+import java.time.LocalTime
+import java.util.UUID
+
+@Deprecated(message = "Can be removed when migration is completed")
+class MigrateBookingIntegrationTest : IntegrationTestBase() {
+
+  @Autowired
+  private lateinit var inboundEventsService: InboundEventsService
+
+  @Autowired
+  private lateinit var videoBookingRepository: VideoBookingRepository
+
+  @Autowired
+  private lateinit var bookingHistoryRepository: BookingHistoryRepository
+
+  @Autowired
+  private lateinit var videoBookingEventRepository: VideoBookingEventRepository
+
+  @Test
+  @Transactional
+  fun `should migrate court booking`() {
+    whereaboutsApi().stubGetVideoBookingToMigrate(
+      1,
+      VideoBookingMigrateResponse(
+        videoBookingId = 1,
+        offenderBookingId = 1,
+        prisonCode = MOORLAND,
+        courtCode = DERBY_JUSTICE_CENTRE,
+        probation = false,
+        createdBy = "MIGRATION COURT USER",
+        madeByTheCourt = true,
+        comments = "Migrated court comments",
+        pre = AppointmentLocationTimeSlot(1, yesterday(), LocalTime.of(9, 0), LocalTime.of(10, 0)),
+        main = AppointmentLocationTimeSlot(1, yesterday(), LocalTime.of(10, 0), LocalTime.of(11, 0)),
+        post = AppointmentLocationTimeSlot(1, yesterday(), LocalTime.of(11, 0), LocalTime.of(12, 0)),
+        cancelled = false,
+        events = listOf(
+          VideoBookingEvent(
+            eventId = 1,
+            eventTime = yesterday().atStartOfDay(),
+            eventType = VideoLinkBookingEventType.CREATE,
+            comment = "Court booking create event comments",
+            courtCode = DERBY_JUSTICE_CENTRE,
+            preStartTime = yesterday().atTime(9, 0),
+            preEndTime = yesterday().atTime(10, 0),
+            mainStartTime = yesterday().atTime(10, 0),
+            mainEndTime = yesterday().atTime(11, 0),
+            postStartTime = yesterday().atTime(11, 0),
+            postEndTime = yesterday().atTime(12, 0),
+            prisonCode = MOORLAND,
+            createdByUsername = "MIGRATION COURT CREATE USER",
+            courtName = null,
+            madeByTheCourt = true,
+            preLocationId = 1,
+            mainLocationId = 1,
+            postLocationId = 1,
+          ),
+        ),
+      ),
+    )
+
+    prisonSearchApi().stubPostGetPrisonerByBookingId(1, "ABC123", MOORLAND)
+
+    UUID.randomUUID().also { dpsLocationId ->
+      nomisMappingApi().stubGetNomisToDpsLocationMapping(1, dpsLocationId.toString())
+      locationsInsidePrisonApi().stubGetLocationById(dpsLocationId, moorlandLocation)
+    }
+
+    videoBookingRepository.findAll().filter(VideoBooking::isCourtBooking) hasSize 0
+
+    inboundEventsService.process(MigrateVideoBookingEvent(1))
+
+    val migratedCourtBooking = videoBookingRepository.findAll().single(VideoBooking::isCourtBooking)
+
+    with(migratedCourtBooking) {
+      bookingType isEqualTo "COURT"
+      statusCode isEqualTo StatusCode.ACTIVE
+      hearingType isEqualTo "UNKNOWN"
```

Comment:
Still need to sort out the data to back up 'UNKNOWN' if that is what we are going with.